op-batcher/batcher/driver.go

```diff
@@ -407,6 +408,45 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 	}
 }
 
+func (l *BatchSubmitter) waitNodeSyncAndClearState() {
+	// Wait for any in flight transactions
+	// to be ingested by the node before
+	// we start loading blocks again.
+	err := l.waitNodeSync()
+	if err != nil {
+		l.Log.Warn("error waiting for node sync", "err", err)
+	}
+	l.clearState(l.shutdownCtx)
+}
+
+// checkExpectedProgress uses the supplied syncStatus to infer
+// whether the node providing the status has made the expected
+// safe head progress given fully submitted channels held in
+// state.
+func (l *BatchSubmitter) checkExpectedProgress(syncStatus eth.SyncStatus) error {
+
+	// Define a buffer: a delay denominated in L1 blocks
+	// to allow verifier to process blocks.
+	// Set to 0 to require immediate safe head advancement.
+	// Set to 1 to allow e.g. 12s for the safe head to advance
+	// after the L1 inclusion block was ingested.
+	// TODO extract this into a config variable
+	// TODO should we also wait numConfirmations (this is a txmgr config variable, not in scope of the batch submitter)
+
+	BUFFER := uint64(1)
```

Comment:
Just a thought - given how this is used, I wonder whether naming it`PROCESSING_WINDOW` would make it more intuitive?

Reply:
+1 for choosing a better name. What do you think about just using `VERIFIER_L1_CONFS`, which is the flag for the verifier?



https://github.com/ethereum-optimism/optimism/blob/466fd958f7f2ba25c982f9326a9af90aeea1feb7/op-node/flags/flags.go#L206-L212

Reply:
Ah yeah, that sounds great!

Reply:
[cd05cb5](https://github.com/ethereum-optimism/optimism/pull/12430/commits/cd05cb5be1bb2ee9c4ce313151f6e849fb138131)